pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -186,10 +182,43 @@ func (w *watcher) run(ctx context.Context, repo git.Repo, repoCfg config.PipedRe
 			}
 			cfg, err := config.LoadEventWatcher(repo.GetPath(), includedCfgs, excludedCfgs)
 			if errors.Is(err, config.ErrNotFound) {
-				w.logger.Info("configuration file for Event Watcher not found",
+				w.logger.Info("configuration file for Event Watcher in .pipe/ not found",
 					zap.String("repo-id", repoCfg.RepoID),
 					zap.Error(err),
 				)
+				resp, err := w.apiClient.ListApplications(ctx, &pipedservice.ListApplicationsRequest{})
+				if err != nil {
+					w.logger.Error("failed to list registered application", zap.Error(err))
+					continue
+				}
+				cfgs := make([]*config.GenericApplicationSpec, 0, len(resp.Applications))
+				for _, app := range resp.Applications {
+					if app.GitPath.Repo.Id != repoCfg.RepoID {
+						continue
+					}
+					cfg, err := config.LoadApplication(repo.GetPath(), app.GitPath.GetApplicationConfigFilePath(), app.Kind)
+					if err != nil {
+						w.logger.Error("failed to load application configuration", zap.Error(err))
+						continue
+					}
+					cfgs = append(cfgs, cfg)
+				}
+				for _, cfg := range cfgs {
+					if cfg.EventWatcher == nil {
+						w.logger.Info("configuration for Event Watcher in application configuration not found",
+							zap.String("repo-id", repoCfg.RepoID),
+							zap.String("app-name", cfg.Name),
+						)
+						continue
+					}
+					if err := w.execute(ctx, repo, repoCfg.RepoID, cfg.EventWatcher); err != nil {
+						w.logger.Error("failed to execute the event from application configuration",
+							zap.String("repo-id", repoCfg.RepoID),
+							zap.Error(err),
+						)
+						continue
+					}
```

Comment:
why don't we do this in the above for-loop?

Reply:
This is so that if .pipe/ does not exist in a particular repository, it will refer to the application configuration to handle some event watcher.
Finally, the process loading the .pipe/ file will be removed.

Reply:
I noticed this is not a good method and has been modified to check both configs every time for ease of transition.
So PTAL when you have enough time.